### PR TITLE
r4 immunization model

### PR DESCRIPTION
**Pull request by ilaflamme-va**
https://vajira.max.gov/secure/RapidBoard.jspa?rapidView=479&projectKey=API&view=detail&selectedIssue=API-809

### CODE REVIEW

File: us-core-r4-examples/src/main/java/gov/va/api/health/uscorer4/api/swaggerexamples/SwaggerImmunization.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+  /**
+   * An example Immunization.
+   *
+   * @return an example Immunization.
+   */
+  public static Immunization immunization() {
+    return Immunization.builder()
+        .resourceType("Immunization")
+        .id("1fd82e3a-a95b-5c04-9a68-c8ddf740ea0c")
```
Comment (jhulbert-va) [inline us-core-r4-examples/src/main/java/gov/va/api/health/uscorer4/api/swaggerexamples/SwaggerImmunization.java:22]: V2 Identity Plz.

File: us-core-r4-examples/src/main/java/gov/va/api/health/uscorer4/api/swaggerexamples/SwaggerImmunization.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        .vaccineCode(
+            CodeableConcept.builder()
+                .text("meningococcal MCV4P")
+                .coding(
+                    asList(
+                        Coding.builder().system("http://hl7.org/fhir/sid/cvx").code("114").build()))
+                .build())
+        .patient(
+            Reference.builder()
+                .reference("https://sandbox-api.va.gov/services/argonaut/v0/Patient/2000163")
```
Comment (jhulbert-va) [inline us-core-r4-examples/src/main/java/gov/va/api/health/uscorer4/api/swaggerexamples/SwaggerImmunization.java:34]: This patient read should match the ICN in the bundle below.

File: us-core-r4/src/main/java/gov/va/api/health/uscorer4/api/resources/Immunization.java [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+
+  @Pattern(regexp = Fhir.DATETIME)
+  String occurrenceDateTime;
+
+  String occurrenceString;
+
+  @Pattern(regexp = Fhir.DATETIME)
+  String recorded;
+
+  Boolean primarySource;
```
Comment (jhulbert-va) [inline us-core-r4/src/main/java/gov/va/api/health/uscorer4/api/resources/Immunization.java:103]: NotNull

File: us-core-r4/src/main/java/gov/va/api/health/uscorer4/api/resources/Immunization.java [hunk 4]
```diff
@@ -0,0 +1,321 @@
...
+
+  @Valid SimpleQuantity doesQuantity;
+
+  @Valid List<Performer> performer;
+
+  @Valid List<Annotation> note;
+
+  @Valid CodeableConcept reasonCode;
+
+  @Valid Reference reasonReference;
```
Comment (jhulbert-va) [inline us-core-r4/src/main/java/gov/va/api/health/uscorer4/api/resources/Immunization.java:128]: List

File: us-core-r4/src/main/java/gov/va/api/health/uscorer4/api/resources/Immunization.java [hunk 5]
```diff
@@ -0,0 +1,321 @@
...
+
+  @Valid Boolean isSubpotent;
```
Comment (jhulbert-va) [inline us-core-r4/src/main/java/gov/va/api/health/uscorer4/api/resources/Immunization.java:130]: Boolean doesnt need valid.

File: us-core-r4/src/main/java/gov/va/api/health/uscorer4/api/resources/Immunization.java [hunk 6]
```diff
@@ -0,0 +1,321 @@
...
+    @Valid List<Extension> extension;
+
+    @Valid List<Extension> modifierExtension;
+
+    @Pattern(regexp = Fhir.DATETIME)
+    String date;
+
+    @Valid Reference detail;
+
+    @Valid Boolean reported;
```
Comment (jhulbert-va) [inline us-core-r4/src/main/java/gov/va/api/health/uscorer4/api/resources/Immunization.java:277]: Boolean doesnt require Valid annotation

File: us-core-r4-examples/src/main/java/gov/va/api/health/uscorer4/api/swaggerexamples/SwaggerImmunization.java [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
```
Comment (jhulbert-va) [inline us-core-r4-examples/src/main/java/gov/va/api/health/uscorer4/api/swaggerexamples/SwaggerImmunization.java:34]: This should be the same ICN as the bundle:
```
BundleLink.builder()
                    .relation(BundleLink.LinkRelation.self)
                    .url(
                        "https://sandbox-api.va.gov/services/argonaut/v0/Immunization?patient=1017283148V813263&page=1&_count=15")
                    .build()
```

Comment (jblefkove-va) [inline us-core-r4-examples/src/main/java/gov/va/api/health/uscorer4/api/swaggerexamples/SwaggerImmunization.java:34]: url should be `services/fhir/v0/r4/`

File: us-core-r4/src/main/java/gov/va/api/health/uscorer4/api/ImmunizationApi.java [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+          "https://build.fhir.org/ig/HL7/US-Core-R4/StructureDefinition-us-core-immunization.html",
+      tags = {"Immunization"})
+  @GET
+  @Path("Immunization/{id}")
+  @ApiResponse(
+      responseCode = "200",
+      description = "Record found",
+      content =
+          @Content(
+              mediaType = "application/json+fhir",
```
Comment (jblefkove-va) [inline us-core-r4/src/main/java/gov/va/api/health/uscorer4/api/ImmunizationApi.java:28]: fhir+json

File: us-core-r4/src/main/java/gov/va/api/health/uscorer4/api/resources/Immunization.java [hunk 9]
```diff
@@ -0,0 +1,321 @@
...
+  @Data
+  @NoArgsConstructor
+  @EqualsAndHashCode(callSuper = true)
+  @JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
+  @JsonDeserialize(builder = Immunization.Bundle.BundleBuilder.class)
+  @Schema(
+      name = "ImmunizationBundle",
+      example =
+          "${r4.immunizationBundle:gov.va.api.health."
+              + "r4.api.swaggerexamples.SwaggerImmunization#immunizationBundle}")
```
Comment (jblefkove-va) [inline us-core-r4/src/main/java/gov/va/api/health/uscorer4/api/resources/Immunization.java:161]: wrong package?

File: us-core-r4/src/main/java/gov/va/api/health/uscorer4/api/resources/Immunization.java [hunk 10]
```diff
@@ -0,0 +1,321 @@
...
+
+    @Valid List<Extension> modifierExtension;
+
+    String series;
+
+    @Valid Reference authority;
+
+    @Valid List<CodeableConcept> targetDisease;
+
+    @Pattern(regexp = Fhir.POSITIVE_INT)
```
Comment (jblefkove-va) [inline us-core-r4/src/main/java/gov/va/api/health/uscorer4/api/resources/Immunization.java:311]: `@Min(1) Integer` ?

Comment (ilaflamme-va) [reply us-core-r4/src/main/java/gov/va/api/health/uscorer4/api/resources/Immunization.java:311]: I changed all of the `Fhir.POSITIVE_INT` regex's to now be `@Min(1) Integer`

**Pull request merged**